source/adiabatic_conditions/initial_profile.cc

```diff
@@ -187,23 +201,80 @@ namespace aspect
     {
       const double z = this->get_geometry_model().depth(p);
 
+      Assert (z >= 0, ExcInternalError());
```

Comment:
Isn't this too strict? Maybe we should allow being eps below 0. You catch this case below anyways (by returning temperatures.front).